tests/max_encoded_len.rs

```diff
@@ -65,18 +65,17 @@ fn generic_max_length() {
 }
 
 #[derive(Encode, MaxEncodedLen)]
-struct CompactField {
+struct CompactField<T: MaxEncodedLen> {
 	#[codec(compact)]
-	t: u64,
+	t: T,
 	v: u64,
 }
 
 #[test]
 fn compact_field_max_length() {
-	assert_eq!(
-		CompactField::max_encoded_len(),
-		Compact::<u64>::max_encoded_len() + u64::max_encoded_len()
-	);
+	let max_len = CompactField::<u64>::max_encoded_len();
+	assert_eq!(max_len, Compact::<u64>::max_encoded_len() + u64::max_encoded_len());
+	assert_eq!(max_len, 17);
 }
```

Comment:
I think I'd prefer to have a separate test for this, instead of modifying an existing one.

Reply:
Yes

Reply:
done, this test would fail with

```
  Compiling parity-scale-codec v3.6.6 (/Users/pg/github/parity-scale-codec)                                                                                                                                                                                            
error[E0277]: the trait bound `Compact<T>: MaxEncodedLen` is not satisfied                                                                                                                                                                                              
  --> tests/max_encoded_len.rs:87:5                                                                                                                                                                                                                                     
   |                                                                                                                                                                                                                                                                    
87 |     t: T,                                                                                                                                                                                                                                                          
   |        ^ the trait `MaxEncodedLen` is not implemented for `Compact<T>`                                                                                                                                                                                             
   |                                                                                                                                                                                                                                                                    
help: consider extending the `where` clause, but there might be an alternative better way to express this requirement                                                                                                                                                   
   |                                                                                                                                                                                                                                                                    
84 | #[derive(Encode, MaxEncodedLen, Compact<T>: MaxEncodedLen)]                                                                    
   |                               +++++++++++++++++++++++++++
```